typings/index.d.ts

```diff
@@ -1,4 +1,2226 @@
+  MAXIMUM_INVITES: 30016;
+  GUILD_ALREADY_HAS_TEMPLATE: 30031;
+  UNAUTHORIZED: 40001;
+  ACCOUNT_VERIFICATION_REQUIRED: 40002;
+  REQUEST_ENTITY_TOO_LARGE: 40005;
+  FEATURE_TEMPORARILY_DISABLED: 40006;
+  USER_BANNED: 40007;
+  ALREADY_CROSSPOSTED: 40033;
+  MISSING_ACCESS: 50001;
+  INVALID_ACCOUNT_TYPE: 50002;
+  CANNOT_EXECUTE_ON_DM: 50003;
+  EMBED_DISABLED: 50004;
+  CANNOT_EDIT_MESSAGE_BY_OTHER: 50005;
+  CANNOT_SEND_EMPTY_MESSAGE: 50006;
+  CANNOT_MESSAGE_USER: 50007;
+  CANNOT_SEND_MESSAGES_IN_VOICE_CHANNEL: 50008;
+  CHANNEL_VERIFICATION_LEVEL_TOO_HIGH: 50009;
+  OAUTH2_APPLICATION_BOT_ABSENT: 50010;
+  MAXIMUM_OAUTH2_APPLICATIONS: 50011;
+  INVALID_OAUTH_STATE: 50012;
+  MISSING_PERMISSIONS: 50013;
+  INVALID_AUTHENTICATION_TOKEN: 50014;
+  NOTE_TOO_LONG: 50015;
+  INVALID_BULK_DELETE_QUANTITY: 50016;
+  CANNOT_PIN_MESSAGE_IN_OTHER_CHANNEL: 50019;
+  INVALID_OR_TAKEN_INVITE_CODE: 50020;
+  CANNOT_EXECUTE_ON_SYSTEM_MESSAGE: 50021;
+  INVALID_OAUTH_TOKEN: 50025;
+  BULK_DELETE_MESSAGE_TOO_OLD: 50034;
+  INVALID_FORM_BODY: 50035;
+  INVITE_ACCEPTED_TO_GUILD_NOT_CONTAINING_BOT: 50036;
+  INVALID_API_VERSION: 50041;
+  CANNOT_DELETE_COMMUNITY_REQUIRED_CHANNEL: 50074;
+  REACTION_BLOCKED: 90001;
+  RESOURCE_OVERLOADED: 130000;
+}
+
+export type APIMessageContentResolvable = string | number | boolean | bigint | symbol | readonly StringResolvable[];
+
+export interface ApplicationAsset {
+  name: string;
+  id: Snowflake;
+  type: 'BIG' | 'SMALL';
+}
+
+export interface AuditLogChange {
+  key: string;
+  old?: any;
+  new?: any;
+}
+
+export interface AwaitMessagesOptions extends MessageCollectorOptions {
+  errors?: string[];
+}
+
+export interface AwaitReactionsOptions extends ReactionCollectorOptions {
+  errors?: string[];
+}
+
+export interface BanOptions {
+  days?: number;
+  reason?: string;
+}
+
+export type Base64Resolvable = Buffer | Base64String;
+
+export type Base64String = string;
+
+export type BitFieldResolvable<T extends string> =
+  | RecursiveReadonlyArray<T | number | Readonly<BitField<T>>>
+  | T
+  | number
+  | Readonly<BitField<T>>;
+
+export type BufferResolvable = Buffer | string;
+
+export interface ChannelCreationOverwrites {
+  allow?: PermissionResolvable | number;
+  deny?: PermissionResolvable | number;
+  id: RoleResolvable | UserResolvable;
+}
+
+export interface ChannelData {
+  name?: string;
+  position?: number;
+  topic?: string;
+  nsfw?: boolean;
+  bitrate?: number;
+  userLimit?: number;
+  parentID?: Snowflake | null;
+  rateLimitPerUser?: number;
+  lockPermissions?: boolean;
+  permissionOverwrites?: readonly OverwriteResolvable[] | Collection<Snowflake, OverwriteResolvable>;
+}
+
+export interface ChannelLogsQueryOptions {
+  limit?: number;
+  before?: Snowflake;
+  after?: Snowflake;
+  around?: Snowflake;
+}
+
+export interface ChannelPosition {
+  channel: ChannelResolvable;
+  position: number;
+}
+
+export type ChannelResolvable = Channel | Snowflake;
+
+export enum ChannelType {
```

Comment:
`ChannelType` shouldn't be exported, it's a private type made for convenience. It also does not exist as a JSDoc but as the description of [`Channel#type`](https://discord.js.org/#/docs/main/stable/class/Channel?scrollTo=type), therefore, it shouldn't be exported.

Reply:
You cannot declare enums in declaration files without exporting them, so it's either convert to a type (could do that here) or actually create an export in the source (would probably need another `fix:` pr)

Reply:
Can you share the error you get by doing `declare enum ChannelType`? That's also what compiled TS code does with enums (it does `export declare enum <name>`)?

Reply:
It doesn't give an error but you can still import the enum that way